### PR TITLE
Updates pipelines to compress flow

### PR DESCRIPTION
**Pull request by jbeilstein**
Updated pipelines.py
--now compresses flow if outlet pressure is negative
--calculates CO2 generated and additional cost of adding compressors

Updated facilities.py
--added pipeline compressor costs to facilities cost

Updated main.py
--linked outlet pressure of pipeline to input of facilities

Updated main_effects.py
--added code to calculate main effects on CAPEX.

### CODE REVIEW

File: Assign2/main.py [hunk 1]
```diff
@@ -280,7 +283,7 @@ def experiment(injection_tubing_diameter, num_wells, num_connections, mass_flow_
 
 
 # TO DO: Add in the DOE generation functionality. 
-imported_df = pd.read_csv("./file_import_and_graph/DOE_tanner.csv", index_col=0)
+imported_df = pd.read_csv("C:/Users/Beerstein/GitHub/MDO_Project/Assign2/file_import_and_graph/DOE_tanner.csv", index_col=0)
```
Comment (tjpapenfuss) [inline Assign2/main.py]: can you revert this change back to the original? This might cause others code to fail. "./file_import_and_graph/DOE_tanner.csv"
```suggestion
imported_df = pd.read_csv("./file_import_and_graph/DOE_tanner.csv", index_col=0)
```

File: Assign2/file_import_and_graph/main_effect.py [hunk 2]
```diff
@@ -19,21 +20,29 @@ def main_effect(dict,var):
     import DOE_generator as deo_gen
 
     #Change the filepath in this line to reflect where the "DOE_options_only.csv" resides.
-    key_list,val_list,df = deo_gen.DOE_generator('C:/Users/jbeil/MDO_Project/Assign2/file_import_and_graph/DOE_options_only.csv')
```
Comment (tjpapenfuss) [inline Assign2/file_import_and_graph/main_effect.py:22]: Same here. Can we get rid of "C:/Users/jbeil/MDO_Project/Assign2"

```suggestion
    key_list,val_list,df = deo_gen.DOE_generator('./file_import_and_graph/DOE_options_only.csv')
```

File: Assign2/file_import_and_graph/main_effect.py [hunk 3]
```diff
@@ -19,21 +20,29 @@ def main_effect(dict,var):
...
 
-    output = key_list[8]
-    df[output] = df[output].astype('int32')
+    key_list,val_list,df = deo_gen.DOE_generator('C:/Users/Beerstein/GitHub/MDO_Project/Assign2/file_import_and_graph/DOE_options_only.csv')
```
Comment (tjpapenfuss) [inline Assign2/file_import_and_graph/main_effect.py]: see above comment.

**Review by tjpapenfuss — CHANGES_REQUESTED** [review]

File: Assign2/main.py (hunk 1 above)
Comment (onitsuka-tiger) [reply Assign2/main.py:286]: support tanner's suggestion.

File: Assign2/facilities.py
Comment (onitsuka-tiger) [inline Assign2/facilities.py:1]: THe changes in facilities.py look good. I don't see the CO2_emit_pipes carried through in the main.py though? need to add that variable into the inputs of the fxn for it to work.

File: Assign2/main.py [hunk 4]
```diff
@@ -41,7 +43,8 @@ def experiment(injection_tubing_diameter, num_wells, num_connections, mass_flow_
     # ------------------------------------------------------------------------------------------------------------------ #
 
     ## 12 Compressor ------------------------------------------------------------------------------##
-    press_out = 345 # pipes_press_out * 6.89476  #
+    #press_out = 345 
+    press_out = pipes_press_out# * 6.89476  #
     p2 = 700 #kPa
```
Comment (onitsuka-tiger) [inline Assign2/main.py]: update p2  to "=press_out"

File: Assign2/main.py [hunk 5]
```diff
@@ -41,7 +43,8 @@ def experiment(injection_tubing_diameter, num_wells, num_connections, mass_flow_
...
     test_temp_out = pipes_temp_out # 300
```
Comment (onitsuka-tiger) [inline Assign2/main.py:49]: just confirming if we need to convert to kelvin here or not. If so, convert, if not its fine as is.

**Review by onitsuka-tiger — COMMENTED** [review]
Overall looks good from pipe to fac. Looking at the temp_out, just wanted to make sure to include the appropriate conversion if not kelvin.

File: Assign2/main.py (hunk 4 above)
Comment (onitsuka-tiger) [reply Assign2/main.py]: not necessary anymore?

File: Assign2/file_import_and_graph/main_effect.py (hunk 3 above)
Comment (tjpapenfuss) [inline Assign2/file_import_and_graph/main_effect.py]: can we update this?

File: Assign2/main.py (hunk 1 above)
Comment (tjpapenfuss) [inline Assign2/main.py]: change

File: Assign2/main.py (hunk 4 above)
Comment (jbeilstein) [reply Assign2/main.py]: Complied with

File: Assign2/main.py (hunk 1 above)
Comment (jbeilstein) [reply Assign2/main.py]: Fixed

File: Assign2/file_import_and_graph/main_effect.py (hunk 2 above)
Comment (jbeilstein) [reply Assign2/file_import_and_graph/main_effect.py:22]: Fixed

File: Assign2/file_import_and_graph/main_effect.py (hunk 3 above)
Comment (jbeilstein) [reply Assign2/file_import_and_graph/main_effect.py]: Fixed

File: Assign2/main.py (hunk 5 above)
Comment (jbeilstein) [reply Assign2/main.py:49]: Output is set to kelvin

**Review by tjpapenfuss — APPROVED** [review]

**Review by onitsuka-tiger — COMMENTED** [review]
I'm good with the facilities.ph changes.
